### ISSUE TITLE
Inline loading component should be an aria-live region

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#1444]
## Detailed description

By adding the attributes `role="alert"` and `aria-live="assertive"` and creating some affordances for changing that markup in our JS visually impaired users with screen reading software can have the full intended experience.

Currently the users with screen readers have no announcements of status read to them.

## Additional information

This was tested using screen reader on macOS.
Some helpful links for implementing functional live regions:
[How To Create a Simple CSS Loading Spinner and Make it Accessible](https://codeburst.io/how-to-create-a-simple-css-loading-spinner-make-it-accessible-e5c83c2e464c)
[It's Alive!: Apps That Feedback Accessibly](https://www.smashingmagazine.com/2015/04/its-alive-apps-that-feed-back-accessibly/)
